Add some comments and types

tonycaputome opened the pull request:
- Rename model folder to types.
- Add some comment.
- Change tsconfig allows TS look for a index file in a folder.

tonycaputome committed:
refactor: Add some comments and types

- Rename model folder to types
- Add some comment
- Change tsconfig allows TS look for a index file in a folder.

Cadienvan commented on src/index.ts:
```diff
@@ -1,11 +1,20 @@
-
+import { arraify, isPromise } from "./lib";
+import { CacheItem, TKey, TMapCache, TStrings } from "./types";
+
```
I completely agree with you, still I don't have the knowledge to provide a working example. Please, remove the comment or change it as a "@todo"

Cadienvan commented on src/index.ts:
```diff
@@ -14,33 +23,31 @@ export class KeyValueCache {
   exec(
-    fn: Function,
-    key: string | Array<string>,
+    fn: () => Promise<any>,
```
It doesn't necessarily have to be a promise, you can pass a long running sync function. Please, revert to a sync function or bring both sync and async to the table.

Cadienvan commented on src/index.ts:
```diff
@@ -14,33 +23,31 @@ export class KeyValueCache {
+    dependencyKeys: TKey = [],
     ttl = this.DEFAULT_TTL
-  ): unknown {
+  ): Pick<CacheItem, "value"> | Promise<any> {
```
Thank you, that was exactly the static typing I was looking for!

Cadienvan commented on src/index.ts:
```diff
@@ -50,33 +57,38 @@ export class KeyValueCache {
-  get(key: string | Array<string>) {
-    const _keys = arraify(key)
-    const cacheDataItem = this.#appCache.get(this.#getMapKey(_keys));
+  isCached = (key: TKey) => {
```
Not sure about this method, prepending with an "is" is supposedly expected to return a boolean, but this is not. Still, not gonna argue about that, just wanted some discussion.

tonycaputome replied:
You are right, usually prepending "is" means we are waiting a return value as boolean. I just created this method due the replicated code. Just that
```ts
const _keys = arraify(key);
return this.#appCache.get(this.#getMapKey(_keys));
```

Cadienvan commented on src/index.ts:
```diff
@@ -138,16 +148,17 @@ export class KeyValueCache {
 
-  invalidateByKeys(keys: Array<string | RegExp>): number {
-    return keys.reduce((acc, key) => acc + this.invalidateByKey(key), 0);
+  invalidateByKeys(keys: Array<string | RegExp>) {
```
A counter could be useful for debugging purposes, who knows what the use case could be. Please, revert back to returning the sum.

tonycaputome replied:
No problem.

Cadienvan reviewed (changes requested):
Thank you, I just put a couple things to consider / revert, but really a lot of thanks for the invaluable help!

tonycaputome committed:
refactor: Discussing the code review

- Add a todo block
- invalidateByKeys method return a number using the .reduce method
- renamed method isCache to a simple "cache"

Cadienvan commented:
Thanks a lot!
